### PR TITLE
Private entry fields

### PR DESCRIPTION
**Pull request by AljoschaMeyer**
I did not add methods to mutate fields yet. We can add those *if* the need arises.

Not only privatises `Entry`, but also `Range3d` and `Area`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,3 +3,5 @@
 members = ["data-model", "earthstar", "fuzz"]
 resolver = "2"
 
+[workspace.lints.clippy]
+type_complexity = "allow"
```
Comment (sgwilym) [inline Cargo.toml:7]: _thank you_

File: data-model/src/encoding/relativity.rs [hunk 2]
```diff
@@ -283,51 +284,51 @@ where
         let namespace_id = if is_namespace_encoded {
             N::decode(producer).await?
         } else {
-            reference.namespace_id.clone()
+            reference.get_namespace_id().clone()
```
Comment (sgwilym) [inline data-model/src/encoding/relativity.rs:287]: is `get_whatever` really the norm for this kind of thing? It's so unwieldy compared to `namespace_id` :(

**Comment by sgwilym** [comment]
What does making these fields private get us? To make it impossible to mutate fields after construction?

**Comment by AljoschaMeyer** [comment]
That is not the primary reason. It allows us to later change the definition of the types. We might have an enum where one variant looks like the current struct, but another variant is simply a pointer into a refcounted buffer that contains an encoded version we got from somewhere. Or for `Entry`, we might decide to group path, subspace_id and timestamp together in a substruct that can be used for addressing things in a store.

By using public fields, all those options would require a breaking change.

File: data-model/src/encoding/relativity.rs (hunk 2 above)
Comment (AljoschaMeyer) [reply data-model/src/encoding/relativity.rs:287]: I had `namespace_id_as_ref` before, and then came to my senses =P

Comment (AljoschaMeyer) [reply data-model/src/encoding/relativity.rs:287]: I can also live with `fn namespace_id` though.

Comment (sgwilym) [reply data-model/src/encoding/relativity.rs:287]: That would resolve my only resistance to this change :)

Comment (AljoschaMeyer) [reply data-model/src/encoding/relativity.rs:287]: Ok, on it.

Comment (AljoschaMeyer) [reply data-model/src/encoding/relativity.rs]: Done

**Pull request merged**